cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/bit.cuh>
+
+namespace cudf {
+
+namespace detail {
+
+/**---------------------------------------------------------------------------*
+ * @brief An immutable, non-owning view of device data as a column of elements
+ * that is trivially copyable and usable CUDA device code.
+ *---------------------------------------------------------------------------**/
+class alignas(16) column_device_view_base {
```

Comment:
Why do we need `alignas(16)`?

Reply:
Technically we don't _need_ it. I added it because there will be arrays of `column_view_view_base` objects (i.e., a `device_table_view`) in device memory, and I wanted to guarantee aligned accesses to objects in that array.